src/python/pants/backend/terraform/goals/check.py

```diff
@@ -41,14 +45,26 @@ async def terraform_check(
     )
     files_by_directory = partition_files_by_directory(source_files.files)
 
+    fetched_deps = await Get(
+        TerraformDependencies,
+        GetTerraformDependenciesRequest(source_files, tuple(files_by_directory.keys())),
+    )
+    # just merge them all for now. This will probably be a problem with multiple TF sources requesting different versions of the same providers
```

Comment:
Isn't that a `terraform`-level problem though? I.e., wouldn't that be an issue even without Pants?

Reply:
Pants might introduce another way for it to be a problem, in that it looks like it selects multiple terraform_modules and includes them all in the same (single) partition. Essentially the same as if Python didn't have the concept of resolves: it would be possible for different Python sources to depend on different versions of dependencies.
I think terraform does the right thing here because we use the `-chdir` argument, so even though all the sources are included terraform is only looking in the correct subdir. I'm not sure though

Reply:
Gotcha. I suppose we could run `tf init` on each directory in a separate process? But that can be a followup investigation.